nemo/core/neural_factory.py

```diff
@@ -36,10 +37,117 @@
 from ..utils import ExpManager
 from .callbacks import ActionCallback, EvaluatorCallback
 from .neural_types import *
+from nemo.utils.app_state import AppState
 from nemo.utils.decorators import deprecated
 
 logging = nemo.logging
 
+# def topological_sort_from_leaves(leaf_nmtensors, cached_training_state: TrainingState = None):
+def topological_sort_from_leaves(leaf_nmtensors, cached_training_state=None):
```

Comment:
? this looks like a copy of inverse graph traversing from actions, am I right?

Reply:
It has been removed from actions

Reply:
but.. why you put it in here??

Reply:
It is now backend agnostic (when datalayer check is fixed). So here is a better place for it than pytorch/actions

Reply:
? but... please correct me - but this code has **nothing** to do with NeuralModuleFactory...?

Reply:
maybe I am missing something, I will discuss it with @blisc  directly